pkgs/shells/zsh/zsh-abbr/default.nix

```diff
@@ -17,9 +17,11 @@ stdenv.mkDerivation rec {
   strictDeps = true;
 
   installPhase = ''
-    install -D zsh-abbr.zsh $out/share/zsh/${pname}/abbr.plugin.zsh
-    # Needed so that `man` can find the manpage, since it looks via PATH
-    mkdir -p $out/bin
+    install -D zsh-abbr.plugin.zsh $out/share/zsh/${pname}/zsh-abbr.plugin.zsh
+    install -D zsh-abbr.zsh        $out/share/zsh/${pname}/zsh-abbr.zsh
+    install -D completions/_abbr   $out/share/zsh/${pname}/completions/_abbr
```

Comment:
Nit: (Untested.) 

```suggestion
    install zsh-abbr.plugin.zsh zsh-abbr.zsh completions/_abbr -Dt $out/share/zsh/${pname}/
```

Reply:
This does not work since `fpath+=${0:A:h}/completions` (`zsh-abbr.plugin.zsh`) will not work and a patching has to be done. But I can try to reduce the two first lines into one as you'd suggested.